spec/controllers/user/swaps_controller_spec.rb

```diff
@@ -71,26 +71,183 @@
       end
 
       describe "PUT #update" do
-        it "confirms the swap if all ducks are lined up" do
-          swap = Swap.create(chosen_user_id: swap_user.id)
-          new_user.incoming_swap = swap
-          swap_user.outgoing_swap = swap
-
-          expect(swap_user.swap.confirmed).to be nil
-
-          put :update, params: { swap: { confirmed: true } }
-
-          expect(response).to redirect_to :user
-          expect(swap_user.swap.chosen_user_id).to eq new_user.id
-          expect(swap_user.swap.confirmed).to be true
+        context "when the user has an INcoming swap" do
+          before do
+            new_user.save! # seemed to be necessary to get all the foreign keys right ... maybe try deleting later
+            new_user.incoming_swap = swap
+            new_user.save! # seemed to be necessary to get all the foreign keys right ... maybe try deleting later
+
+            swap_user.save! # seemed to be necessary to get all the foreign keys right ... maybe try deleting later
+            swap_user.outgoing_swap = swap
+            swap_user.save! # seemed to be necessary to get all the foreign keys right ... maybe try deleting later
```

Comment:
You should be able to get rid of these, if you use this instead: allow(swap).to receive(:choosing_user).and_return(new_user), from memory they probably need to be inside the it / scenario, not at context scope

Reply:
Gah ... I totally forgot about those comments ... I will see if I can do better.